dd-trace/src/telemetry.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    sync::{Arc, Mutex, OnceLock},
+    time::Duration,
+};
+
+use ddtelemetry::{
+    data::{self},
+    worker::{self, TelemetryWorkerHandle},
+};
+
+use crate::{dd_debug, dd_error, dd_info, Config};
+
+static INIT_TELEMETRY_LOCK: Mutex<()> = Mutex::new(());
+
+static TELEMETRY: OnceLock<Arc<Mutex<Telemetry>>> = OnceLock::new();
```

Comment:
I don't think you need both `Mutex` and `OnceLock` 🤔 

The pattern for `OnceLock` is usually

```rust
static TELEMETRY: OnceLock<Arc<Mutex<Telemetry>>> = OnceLock::new();

fn init_telemetry() {
  let Some(telemetry) = TELEMETRY. get_or_init(|| {
    // Create and start the handle here
    // This closure is only run once in the lifetime of the process
    match make_telemetry_worker() {
      Ok(handle) => Some(handle),
      Err(e) => { /* handle err */ ; None }, 
    }
  }) else {
    // there was an error and it has been logged already
    // return
  };
  // Do whatever you need with the handle here
}
```

Reply:
done!